File: chapter1(1-50)/third/prime_factor.py
def is_prime(n):
    if n==1:
        return False
    else:
        for i in range(2,n):
            if n%i==0:
                return False
    return True    
def find_max(a):
    max_num=a[0]
    for i in range(1,len(a)):
        if(a[i]>max_num):
            max_num=a[i] 
    return max_num
list_a=[]
num= 600851475143
for i in range(2,600851475144):
    if(is_prime(i)==True and num%i==0): 
        list_a.append(i)
        num=num//i
result=find_max(list_a)
print(result)
#效率太低，因为我是小白

File: chapter1(1-50)/third/prime_factor.rs
fn is_prime(n:i64)->bool{
    if n==1{
        return false;
    }
    for i in 2..=(n as f64).sqrt() as i64{
        if n%i==0{
            return false;
        }
    }
    true
}

fn main(){
    let  mut num=600851475143;

    let mut vec:Vec<i64>=Vec::new();
    for i in 2..=(num as f64).sqrt() as i64{
        if num%i==0 && is_prime(i)==true{
            vec.push(i);
            num=num/i;
        }
    }
    let max_num=vec.iter().max().unwrap();
    println!("{}",max_num);
}
